src/software/thunderscope/robot_communication.py

```diff
@@ -205,8 +213,9 @@ def __enter__(self):
         self.send_estop_state_thread.start()
         self.run_thread.start()
 
-    def __exit__(self):
+    def __exit__(self, type, value, traceback):
         """Exit RobotCommunication context manager
 
         """
+        self.keep_running = False
         self.run_thread.join()
```

Comment:
Curious why during `__exit__` we join the run thread? From searching online, this seems to be related to the `with` statements, but I can't fully wrap my head around it.